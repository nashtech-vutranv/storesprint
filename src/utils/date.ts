import moment from 'moment'
import {FORMAT_DATE, FORMATE_DATE_NO_HOURS} from '../constants'

export const formatDate = (date: string, format: string = FORMAT_DATE) => {
  return moment(date).format(format)
}

export const formatDateNoHours = (date: string, format: string = FORMATE_DATE_NO_HOURS) => {
  return moment(date).format(format)
}
