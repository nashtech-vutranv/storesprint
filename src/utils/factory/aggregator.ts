import faker from '@faker-js/faker'

export const mockAggregator = () => {
  return {
    id: faker.random.word(),
    name: faker.random.word(),
    version: faker.datatype.number(),
    status: faker.datatype.string(),
  }
}

export const mockAggregators = () => {
  return {
    data: {
      content: Array.apply(null, Array(10)).map(() => mockAggregator()),
      totalElements: 10,
    },
  }
}
