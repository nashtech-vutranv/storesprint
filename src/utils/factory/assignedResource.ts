import faker from '@faker-js/faker'

export const mockAssignedResource = () => {
  return {
    data: {
      createdAt: faker.datatype.string(),
      id: faker.datatype.string(),
      modifiedAt: faker.datatype.string(),
      organizationResponse: {
        additionalInformation: faker.datatype.string(),
        addressLine1: faker.datatype.string(),
        addressLine2: null,
        addressLine3: null,
        city: faker.datatype.string(),
        contactEmailAddress: faker.datatype.string(),
        contactName: faker.datatype.string(),
        contactPhoneNumber: faker.datatype.string(),
        createdAt: faker.datatype.string(),
        erpId: faker.datatype.string(),
        id: faker.datatype.string(),
        modifiedAt: faker.datatype.string(),
        name: faker.datatype.string(),
        postCode: faker.datatype.string(),
        registrationNumber: faker.datatype.string(),
        status: faker.datatype.string(),
        vatRegistrationNumber: faker.datatype.string(),
        version: faker.datatype.number(),
      },
      siteResponse: {
        createdAt: faker.datatype.string(),
        erpId: faker.datatype.string(),
        id: faker.datatype.string(),
        modifiedAt: faker.datatype.string(),
        name: faker.datatype.string(),
        organizationId: faker.datatype.string(),
        status: faker.datatype.string(),
        url: faker.datatype.string(),
        version: faker.datatype.number(),
      },
      userResponse: null,
      version: faker.datatype.number(),
    },
  }
}
