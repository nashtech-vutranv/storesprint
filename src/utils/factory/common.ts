import {IPaginationResponse} from '../../interface'

export const mockOption = () => ({
  value: '',
  label: '',
})

export const mockPaginationResponse = (): IPaginationResponse => {
  return {
    pageable: {
      sort: {
        empty: true,
        sorted: false,
        unsorted: true,
      },
      offset: 0,
      pageSize: 3,
      pageNumber: 0,
      unpaged: false,
      paged: true,
    },
    last: false,
    totalPages: 2,
    totalElements: 5,
    number: 0,
    size: 3,
    first: true,
    numberOfElements: 3,
    sort: {
      empty: true,
      sorted: false,
      unsorted: true,
    },
    empty: false,
  }
}
