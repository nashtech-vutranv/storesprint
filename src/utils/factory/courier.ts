import faker from '@faker-js/faker'
import {ICourierMapping} from '../../interface'

export const mockCourier = (): ICourierMapping => {
  return {
    id: faker.datatype.string(),
    createdAt: faker.datatype.string(),
    modifiedAt: faker.datatype.string(),
    mmsValue: faker.datatype.string(),
    version: faker.datatype.number(),
    mappingTarget: {
        id: faker.datatype.string(),
        mappingTypeId: faker.datatype.string(),
        marketplaceCode: faker.datatype.string(),
        marketplaceTypeId: faker.datatype.string(),
        marketplaceValue: faker.datatype.string(),
    },
    mappingType: {
        id: faker.datatype.string(),
        name: faker.datatype.string(),
    },
    mappingTypeId: faker.datatype.string(),
    marketplaceType: {
        id: faker.datatype.string(),
        name: faker.datatype.string(),
    },
    marketplaceTypeId: faker.datatype.string(),
    marketplaceValue: faker.datatype.string(),
  }
}

export const mockCouriers = (): ICourierMapping[] => {
    return Array.apply(null, Array(10)).map(() =>mockCourier()) 
}
