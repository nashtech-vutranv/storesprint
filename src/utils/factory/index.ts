export * from './organization'
export * from './warehouse'
export * from './aggregator'
export * from './marketplaces'
export * from './currency'
export * from './relationshipAM'
export * from './marketplaceRelationships'
export * from './product'
export * from './order'
export * from './listingStatus'
export * from './common'
export * from './defaultPropertyValue'
export * from './mappings'
export * from './productProperties'
export * from './assignedResource'
export * from './productCategoryMappings'
export * from './productPropertyMappings'
export * from './courier'
export * from './valueMapping'
export * from './courierMappings'
export * from './returnsrefunds'
export * from './deliveryTypeMappings'
