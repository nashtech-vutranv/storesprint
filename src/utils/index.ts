export * from './array'
export * from './layout'
export * from './date'
export * from './conditionPromise'
