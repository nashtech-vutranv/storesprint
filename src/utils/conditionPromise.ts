import {mockOrganizations, mockSites, mockSiteLocales} from './factory/organization'
import {mockMarketplaces, mockCalculateAssignMarketplaces} from './factory/marketplaces'
import {mockAssignedProducts, mockAssignedProductsResponse} from './factory/product'
import {mockWarehouses} from './factory/warehouse'
import {mockMarketplaceRelationship} from './factory/marketplaceRelationships'

type IMockType = 'organization' | 'site' | 'siteLocale' |
 'marketplace' | 'calculateMarketplace' | 'warehouse' 
 | 'marketplaceRelationship' | 'assignedProduct' | 'assignedProductResponse'

export const conditionPromise = (objectFilter: unknown, typeMock: IMockType, objectPaginationQuery?: any) => {
  if (objectFilter || (objectFilter && objectPaginationQuery)) {
    switch (typeMock) {
      case 'organization': return Promise.resolve(mockOrganizations())
      case 'site': return Promise.resolve(mockSites())
      case 'siteLocale': return Promise.resolve(mockSiteLocales())
      case 'marketplace': return Promise.resolve(mockMarketplaces())
      case 'calculateMarketplace': return Promise.resolve(mockCalculateAssignMarketplaces())
      case 'warehouse': return mockWarehouses()
      case 'marketplaceRelationship': return mockMarketplaceRelationship()
      case 'assignedProduct': return Promise.resolve(mockAssignedProducts())
      case 'assignedProductResponse': return Promise.resolve(mockAssignedProductsResponse())

      default: return {}
    }
  } else return Promise.reject()
}