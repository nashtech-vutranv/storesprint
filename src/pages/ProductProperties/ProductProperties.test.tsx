import {render, fireEvent, waitFor} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import '@testing-library/jest-dom'
import {mockProductProperties} from '../../utils/factory'
import {ProductProperties} from '.'

const API = require('../../services/ProductPropertiesService')

jest.mock(API, () => {
  const instance = jest.fn().mockReturnValue({
    getProductProperties: jest.fn().mockImplementation(() =>
      Promise.resolve({
        data: mockProductProperties(),
      })
    ),
  })
  return instance
})

test('renders ProductProperties Page', async () => {
  const tree = render(
    <MemoryRouter>
      <ProductProperties />
    </MemoryRouter>
  )
  await waitFor(() => {
    expect(tree).toMatchSnapshot()
  })
  
  const refresh = tree.container.querySelector(
    '.p-button-icon.pi-refresh'
  ) as Element
  await waitFor(() => {
    expect(refresh).toBeTruthy()
  })
  
  fireEvent.click(refresh)
  await waitFor(() =>
    expect(
      tree.queryByText(/product_properties_page_title/i)
    ).toBeInTheDocument()
  )
})
