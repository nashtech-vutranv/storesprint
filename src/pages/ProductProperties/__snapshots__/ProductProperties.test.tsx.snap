// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ProductProperties Page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            class="page-title-box"
          >
            <h4
              class="page-title"
            >
              product_properties_page_title
            </h4>
          </div>
        </div>
        <div
          class="p-datatable p-component p-datatable-selectable p-datatable-resizable p-datatable-resizable-fit p-datatable-responsive-scroll data-table-mh"
          data-scrollselectors=".p-datatable-wrapper"
          pr_id_1=""
        >
          <div
            class="p-datatable-header"
          >
            <div
              class="row"
            >
              <div
                class="d-flex"
              >
                <div
                  class="d-flex justify-content-start align-items-center w-full"
                >
                  <div
                    class="btn-mr-1 col-xxl-7 col-xl-7 col-lg-7 col-md-8 col-sm-8 col-8"
                  >
                    <span
                      class="p-input-icon-right d-block"
                    >
                      <input
                        class="p-inputtext p-component"
                        placeholder="common_input_search_placeHolder"
                        value=""
                      />
                    </span>
                  </div>
                  <button
                    class="btn-h-95 btn btn-primary"
                    type="button"
                  >
                    common_button_search_label
                  </button>
                </div>
                <div
                  style="display: flex; justify-content: flex-end; width: 100%;"
                />
              </div>
            </div>
          </div>
          <div
            class="p-datatable-wrapper"
          >
            <table
              class="p-datatable-table"
              role="table"
            >
              <thead
                class="p-datatable-thead"
              >
                <tr
                  role="row"
                >
                  <th
                    aria-sort="ascending"
                    class="p-sortable-column p-resizable-column p-highlight"
                    role="columnheader"
                    tabindex="0"
                  >
                    <span
                      class="p-column-resizer"
                    />
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      >
                        product_properties_column_header_erpId
                      </span>
                      <span
                        class="p-sortable-column-icon pi pi-fw pi-sort-amount-up-alt"
                      />
                    </div>
                  </th>
                  <th
                    aria-sort="none"
                    class="p-sortable-column p-resizable-column"
                    role="columnheader"
                    tabindex="0"
                  >
                    <span
                      class="p-column-resizer"
                    />
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      >
                        product_properties_column_header_name
                      </span>
                      <span
                        class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                      />
                    </div>
                  </th>
                  <th
                    aria-sort="none"
                    class="p-sortable-column p-resizable-column"
                    role="columnheader"
                    tabindex="0"
                  >
                    <span
                      class="p-column-resizer"
                    />
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      >
                        product_properties_column_header_type
                      </span>
                      <span
                        class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                      />
                    </div>
                  </th>
                  <th
                    aria-sort="none"
                    class="p-sortable-column p-resizable-column"
                    role="columnheader"
                    tabindex="0"
                  >
                    <span
                      class="p-column-resizer"
                    />
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      >
                        product_properties_column_header_locale_sensitive
                      </span>
                      <span
                        class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                      />
                    </div>
                  </th>
                  <th
                    aria-sort="none"
                    class="p-sortable-column p-resizable-column"
                    role="columnheader"
                    tabindex="0"
                  >
                    <span
                      class="p-column-resizer"
                    />
                    <div
                      class="p-column-header-content"
                    >
                      <span
                        class="p-column-title"
                      >
                        product_properties_column_header_status
                      </span>
                      <span
                        class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                      />
                    </div>
                  </th>
                </tr>
              </thead>
              <tbody
                class="p-datatable-tbody"
              >
                <tr
                  class="p-datatable-emptymessage"
                  role="row"
                >
                  <td
                    colspan="5"
                    role="cell"
                  >
                    product_properties_table_empty_message
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            class="p-paginator p-component p-paginator-bottom table-paginator"
          >
            <div
              class="p-paginator-left-content"
            >
              <button
                aria-label="refresh-button"
                class="p-button p-component p-button-text p-button-icon-only"
                type="button"
              >
                <span
                  class="p-button-icon p-c pi pi-refresh"
                />
                <span
                  class="p-button-label p-c"
                >
                  Â 
                </span>
              </button>
            </div>
            <button
              class="p-paginator-first p-paginator-element p-link p-disabled"
              disabled=""
              type="button"
            >
              <span
                class="p-2"
              >
                organization_paginator_first
              </span>
            </button>
            <button
              class="p-paginator-prev p-paginator-element p-link p-disabled"
              disabled=""
              type="button"
            >
              <span
                class="p-3"
              >
                organization_paginator_previous
              </span>
            </button>
            <span
              class="p-paginator-pages"
            />
            <button
              class="p-paginator-next p-paginator-element p-link p-disabled"
              disabled=""
              type="button"
            >
              <span
                class="p-3"
              >
                organization_paginator_next
              </span>
            </button>
            <button
              class="p-paginator-last p-paginator-element p-link p-disabled"
              disabled=""
              type="button"
            >
              <span
                class="p-2"
              >
                organization_paginator_last
              </span>
            </button>
            <div
              class="p-dropdown p-component p-inputwrapper dropdown-table-input opacity-md p-disabled p-inputwrapper-filled"
            >
              <div
                class="p-hidden-accessible"
              >
                <input
                  aria-haspopup="listbox"
                  aria-label="textbox"
                  disabled=""
                  readonly=""
                  type="text"
                />
              </div>
              <div
                class="p-hidden-accessible p-dropdown-hidden-select"
              >
                <select
                  aria-hidden="true"
                  tabindex="-1"
                >
                  <option
                    value=""
                  />
                  <option
                    value="10"
                  >
                    10
                  </option>
                </select>
              </div>
              <span
                class="p-dropdown-label p-inputtext"
              >
                10
              </span>
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-label="dropdown-trigger"
                class="p-dropdown-trigger"
                role="button"
              >
                <span
                  class="p-dropdown-trigger-icon p-clickable pi pi-chevron-down"
                />
              </div>
            </div>
            <span
              style="user-select: none; text-align: center; margin-left: auto;"
            >
              organization_paginator_show_report
            </span>
            <div
              class="p-paginator-right-content"
            />
          </div>
          <div
            class="p-column-resizer-helper"
            style="display: none;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="page-title-box"
        >
          <h4
            class="page-title"
          >
            product_properties_page_title
          </h4>
        </div>
      </div>
      <div
        class="p-datatable p-component p-datatable-selectable p-datatable-resizable p-datatable-resizable-fit p-datatable-responsive-scroll data-table-mh"
        data-scrollselectors=".p-datatable-wrapper"
        pr_id_1=""
      >
        <div
          class="p-datatable-header"
        >
          <div
            class="row"
          >
            <div
              class="d-flex"
            >
              <div
                class="d-flex justify-content-start align-items-center w-full"
              >
                <div
                  class="btn-mr-1 col-xxl-7 col-xl-7 col-lg-7 col-md-8 col-sm-8 col-8"
                >
                  <span
                    class="p-input-icon-right d-block"
                  >
                    <input
                      class="p-inputtext p-component"
                      placeholder="common_input_search_placeHolder"
                      value=""
                    />
                  </span>
                </div>
                <button
                  class="btn-h-95 btn btn-primary"
                  type="button"
                >
                  common_button_search_label
                </button>
              </div>
              <div
                style="display: flex; justify-content: flex-end; width: 100%;"
              />
            </div>
          </div>
        </div>
        <div
          class="p-datatable-wrapper"
        >
          <table
            class="p-datatable-table"
            role="table"
          >
            <thead
              class="p-datatable-thead"
            >
              <tr
                role="row"
              >
                <th
                  aria-sort="ascending"
                  class="p-sortable-column p-resizable-column p-highlight"
                  role="columnheader"
                  tabindex="0"
                >
                  <span
                    class="p-column-resizer"
                  />
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      product_properties_column_header_erpId
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-amount-up-alt"
                    />
                  </div>
                </th>
                <th
                  aria-sort="none"
                  class="p-sortable-column p-resizable-column"
                  role="columnheader"
                  tabindex="0"
                >
                  <span
                    class="p-column-resizer"
                  />
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      product_properties_column_header_name
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </div>
                </th>
                <th
                  aria-sort="none"
                  class="p-sortable-column p-resizable-column"
                  role="columnheader"
                  tabindex="0"
                >
                  <span
                    class="p-column-resizer"
                  />
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      product_properties_column_header_type
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </div>
                </th>
                <th
                  aria-sort="none"
                  class="p-sortable-column p-resizable-column"
                  role="columnheader"
                  tabindex="0"
                >
                  <span
                    class="p-column-resizer"
                  />
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      product_properties_column_header_locale_sensitive
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </div>
                </th>
                <th
                  aria-sort="none"
                  class="p-sortable-column p-resizable-column"
                  role="columnheader"
                  tabindex="0"
                >
                  <span
                    class="p-column-resizer"
                  />
                  <div
                    class="p-column-header-content"
                  >
                    <span
                      class="p-column-title"
                    >
                      product_properties_column_header_status
                    </span>
                    <span
                      class="p-sortable-column-icon pi pi-fw pi-sort-alt"
                    />
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              class="p-datatable-tbody"
            >
              <tr
                class="p-datatable-emptymessage"
                role="row"
              >
                <td
                  colspan="5"
                  role="cell"
                >
                  product_properties_table_empty_message
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="p-paginator p-component p-paginator-bottom table-paginator"
        >
          <div
            class="p-paginator-left-content"
          >
            <button
              aria-label="refresh-button"
              class="p-button p-component p-button-text p-button-icon-only"
              type="button"
            >
              <span
                class="p-button-icon p-c pi pi-refresh"
              />
              <span
                class="p-button-label p-c"
              >
                Â 
              </span>
            </button>
          </div>
          <button
            class="p-paginator-first p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-2"
            >
              organization_paginator_first
            </span>
          </button>
          <button
            class="p-paginator-prev p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-3"
            >
              organization_paginator_previous
            </span>
          </button>
          <span
            class="p-paginator-pages"
          />
          <button
            class="p-paginator-next p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-3"
            >
              organization_paginator_next
            </span>
          </button>
          <button
            class="p-paginator-last p-paginator-element p-link p-disabled"
            disabled=""
            type="button"
          >
            <span
              class="p-2"
            >
              organization_paginator_last
            </span>
          </button>
          <div
            class="p-dropdown p-component p-inputwrapper dropdown-table-input opacity-md p-disabled p-inputwrapper-filled"
          >
            <div
              class="p-hidden-accessible"
            >
              <input
                aria-haspopup="listbox"
                aria-label="textbox"
                disabled=""
                readonly=""
                type="text"
              />
            </div>
            <div
              class="p-hidden-accessible p-dropdown-hidden-select"
            >
              <select
                aria-hidden="true"
                tabindex="-1"
              >
                <option
                  value=""
                />
                <option
                  value="10"
                >
                  10
                </option>
              </select>
            </div>
            <span
              class="p-dropdown-label p-inputtext"
            >
              10
            </span>
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-label="dropdown-trigger"
              class="p-dropdown-trigger"
              role="button"
            >
              <span
                class="p-dropdown-trigger-icon p-clickable pi pi-chevron-down"
              />
            </div>
          </div>
          <span
            style="user-select: none; text-align: center; margin-left: auto;"
          >
            organization_paginator_show_report
          </span>
          <div
            class="p-paginator-right-content"
          />
        </div>
        <div
          class="p-column-resizer-helper"
          style="display: none;"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
